core/src/test/java/kafka/admin/AdminFenceProducersIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.kafka.server.config.ServerLogConfigs;
+import org.apache.kafka.common.errors.InvalidProducerEpochException;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.errors.TimeoutException;
+
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ClusterTestDefaults(serverProperties = {
+        @ClusterConfigProperty(key = ServerLogConfigs.AUTO_CREATE_TOPICS_ENABLE_CONFIG, value = "false"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_PARTITIONS_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_REPLICATION_FACTOR_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_MIN_ISR_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionStateManagerConfig.TRANSACTIONS_ABORT_TIMED_OUT_TRANSACTION_CLEANUP_INTERVAL_MS_CONFIG, value = "2000")
+})
+@ExtendWith(ClusterTestExtensions.class)
+public class AdminFenceProducersIntegrationTest {
+    private static final String TOPIC_NAME = "mytopic";
+    private static final String TXN_ID = "mytxnid";
+    private static final String INCORRECT_BROKER_PORT = "225";
+    private static final ProducerRecord<byte[], byte[]> RECORD = new ProducerRecord<>(TOPIC_NAME, null, new byte[1]);
+    private final ClusterInstance clusterInstance;
+    private Admin adminClient;
+    private KafkaProducer<byte[], byte[]> producer;
+
+    AdminFenceProducersIntegrationTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+    }
+
+    private KafkaProducer<byte[], byte[]> createProducer() {
+        Properties config = new Properties();
+        config.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers());
+        config.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, TXN_ID);
+        config.put(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, "2000");
+        config.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+        config.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+
+        return new KafkaProducer<>(config);
+    }
+
+    @ClusterTest
+    void testFenceAfterProducerCommit() throws Exception {
+        clusterInstance.createTopic(TOPIC_NAME, 1, (short) 1);
+
+        producer = createProducer();
+        producer.initTransactions();
+        producer.beginTransaction();
+        producer.send(RECORD).get();
+        producer.commitTransaction();
```

Comment:
`KafkaProducer<byte[], byte[]>` implement the `AutoCloseable` interface, so we should use the try-resource to close is more better
```java
        try (KafkaProducer<byte[], byte[]> producer = createProducer()) {
            producer.initTransactions();
            producer.beginTransaction();
            producer.send(RECORD).get();
            producer.commitTransaction();
            // do something
        }
```

Reply:
That's great idea. 
I initially avoided using the auto close syntax to prevent nested try-catch blocks, but it seems this issue can be resolved using assertThrow.